### DISCUSSION TITLE
Unable to build 2.2.4 due to mime_types#get type mismatch

### DISCUSSION THREAD

**Discussion by paddatrapper** [savonet/liquidsoap#3751] [Installation] [answered]
Trying to build liquidsoap for packaging the new version into Debian I get the following error

```
File "src/core/decoder/gstreamer_decoder.ml", line 314, characters 37-51:
314 |       (Decoder.test_file ~log ~mimes:mime_types#get
                                           ^^^^^^^^^^^^^^
Error: This expression has type string list
       but an expression was expected of type Decoder.stream list option
```

ocaml-magic-mime version: 1.3.1
Liquidsoap version: 2.2.4

**Comment by vitoyucepi** [comment, marked as answer]
Hi @paddatrapper,
Try building without gstreamer.

**Reply by paddatrapper** [reply]
While that is an answer, I would prefer not to reduce the feature set of liquidsoap shipped in Debian

**Reply by vitoyucepi** [reply]
I've created a new issue #3752.

**Reply by paddatrapper** [reply]
Thanks!

**Reply by toots** [reply]
I would strongly suggest shipping a package without gstreamer. Its architecture is designed for GUI application and is causing issues such as random freeze at shutdown and all its functionalities are now deprecated in favor of ffmpeg.

**Reply by paddatrapper** [reply]
How do I actually disable gstreamer? Not providing the gstreamer dependency seems no longer to be enough, as it still fails at the same point

**Reply by vitoyucepi** [reply]
@paddatrapper, maybe I am missing something and @toots knows better, but may I review the build config and build log?
Also #3769.

**Reply by toots** [reply]
If you don't have the `gstreamer` opam package installed, liquidsoap should not build with gstreamer.
